src/lib/rule-options.js

```diff
@@ -83,6 +83,7 @@ export const NAMING_CONVENTION_OPTIONS_ANGULAR = [
 ];
 
 export const IMMUTABLE_DATA_OPTIONS = {
+  ignoreMapsAndSets: true,
```

Comment:
This option was added in `eslint-plugin-functional`'s [v9.0.0](https://github.com/eslint-functional/eslint-plugin-functional/releases/tag/v9.0.0). But we also support older versions. I'd be concerned they'd throw because of an unknown option. In which case, we should handle this more defensively.